stack/build.Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM ubuntu:jammy
+
+ARG sources
+ARG packages
+
+RUN echo "$sources" > /etc/apt/sources.list
+
+RUN echo "debconf debconf/frontend select noninteractive" | debconf-set-selections && \
+  export DEBIAN_FRONTEND=noninteractive && \
+  apt-get -y $package_args update && \
+  apt-get -y $package_args upgrade && \
+  apt-get -y $package_args install locales && \
```

Comment:
These lines reference `$package_args`, but that doesn't appear to be defined anywhere. The bionic stack image has the following:

https://github.com/paketo-buildpacks/stacks/blob/c00defb718858eb3724c71addae362c9c047f068/bionic/dockerfile/build/Dockerfile#L5

Reply:
(fixed in the tiny stack with: https://github.com/paketo-buildpacks/jammy-tiny-stack/pull/13/files)

Reply:
I think we can remove the references, rather than populate the arguments. Digging into each argument in more detail:

* `--allow-downgrades` - this allows apt to downgrade the install version of a package without prompting. This seems not great.
* `--allow-remove-essential` - this allows apt to remove core components of the base/essential packages without prompting. This also seems not great.
* `--allow-change-held-packages` - this allows apt to proceed with changing held packages without prompting. This seems moderately bad, but not awful, as I don't think we hold packages.
* `--no-install-recommends` - this instructs apt to not consider recommended packages as a dependency for installing. I'm not sure I have enough context on this to have strong opinions either way.

See: https://manpages.ubuntu.com/manpages/xenial/en/man8/apt-get.8.html

When [this change was first introduced](https://github.com/paketo-buildpacks/stacks/commit/edde18c36e988ad7da926783e49fdf5ab5a8b94b) these args were added without any particular context or extra information, so it's hard to know what the original thinking was.

Reply:
@robdimsdale thanks for adding that context. In reference to `--no-install-recommends`, I'm not sure how a package becomes recommended, but I would imagine we wouldn't want to install any packages except the ones we've defined, so I'd say we probably want this flag

Reply:
@sophiewigmore yes, that makes sense. However it's not obvious to me that we want that feature, though, either. I think we rely on transitive dependencies - I don't think we list out the exhaustive, flattened list of libraries. I think we'd have to investigate this further.

That's why I don't have enough context to suggest including or excluding it.

Reply:
https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends suggests that `--no-install-recommends` is probably a good idea.